### PR TITLE
add tf.linalg.diag func frontend

### PR DESCRIPTION
**Pull request by Sindhuinti**
Close https://github.com/unifyai/ivy/issues/11750

### CODE REVIEW

**Comment by Sindhuinti** [comment]
Close https://github.com/unifyai/ivy/issues/11750

**Comment by Sindhuinti** [comment]
Hey @hirwa-nshuti review this PR.

File: ivy/functional/frontends/tensorflow/linalg.py [hunk 1]
```diff
@@ -243,3 +243,8 @@ def lu_matrix_inverse(lower_upper, perm, validate_args=False, name=None):
 @to_ivy_arrays_and_back
 def einsum(equation, *inputs, **kwargs):
     return tf_frontend.einsum(equation, *inputs, **kwargs)
+
+
+@to_ivy_arrays_and_back
+def diag(x,/,*,k=0,name=None):
+    return ivy.diag(a, k=k)
```
[the remark this answers is not in the document]
Comment (Sindhuinti) [reply ivy/functional/frontends/tensorflow/linalg.py]: required changes made

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @Sindhuinti 

The tests now are failing for all backends would you look into this😊

Thanks

**Comment by hirwa-nshuti** [comment]
> Is there any error in my code regarding test fail ?

Would you run the tests locally to see if they pass on your end?

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Would you fix the merge conflicts to rerun the tests here on CI

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**